### ISSUE TITLE
Autocomplete star syntax

### ISSUE THREAD

**Issue by MichaelRFairhurst** [dart-lang/angular_analyzer_plugin#193]
Consider a large & elaborate ngFor:

```
  <div *ngFor="let item of items; let i = index; trackBy: myFunc"></div>
```

Ideally we should be able to autocomplete every word here except `items` and `i`. That is to say, we should be able to start typing and follow the path:

```
  <div |
  <div *ngFor="|
  <div *ngFor="let item |
  <div *ngFor="let item of |
  <div *ngFor="let item of items; |
  <div *ngFor="let item of items; let i = |
  <div *ngFor="let item of items; let i = index; |
  <div *ngFor="let item of items; let i = index; trackBy: |
```

So far only completing "items" and after "trackBy" work.

We may have to change `converter.dart` to create an AST for templates other than their desugared attributes in order to properly understand what's happening and suggest appropriately.

**Comment by zoechi** [comment]
is this meant as some generic form that works for all structural directives, or is this a special case for `*ngFor`?

**Comment by MichaelRFairhurst** [comment]
Great question!

I'm thinking we go through the extra effort of making this work for all
ngFor like directives, even if they are rare

**Comment by MichaelRFairhurst** [comment]
merged #310
